### PR TITLE
ApplicationControllers

### PR DESCRIPTION
**Pull request by AlakaTahir**
Implementation:
Exposed logic for the below features to controller;
- Leave type
- Access type
- Leave application
- Leave recall

Test:
QA yet to validate

Database Change:
Not required

### CODE REVIEW

File: LeaveApplication.API/Controllers/LeaveApplicationController.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        [HttpDelete("DeleteLeaveApplication")]
+        public async Task<IActionResult> DeleteLeaveApplication(Guid id)
+        {
+            var response = await _leaveApplicationInformationservice.DeleteLeaveApplication(id);
+            return Ok(response);
+        }
+
+
+        [HttpGet("GetApplicationByUserId")]
```
Comment (femhab) [inline LeaveApplication.API/Controllers/LeaveApplicationController.cs:43]: We can make the naming more descriptive. i.e. GetLeaveByUserId or GetByUserId

File: LeaveApplication.API/Controllers/LeaveTypeController.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+        [HttpGet("GetLeaveTypeById")]
+
+        public async Task<IActionResult> GetLeaveTypeById(Guid id)
+        {
+            var response = await _leaveTypeInformationService.GetLeaveTypeById(id);
+            return Ok(response);
+        }
+
+        [HttpGet("GetAll")]
```
Comment (femhab) [inline LeaveApplication.API/Controllers/LeaveTypeController.cs:52]: What is this controller method meant to do?

File: LeaveApplication.Model/ViewModel/ResponseModel/UserInformationResponseModel.cs [hunk 3]
```diff
@@ -10,7 +10,6 @@ public class UserInformationResponseModel
         public string Email { get; set; }
         public string PhoneNumber { get; set; }
         public DateTime CreatedDate { get; set; }
-        public DateTime UpdatedTime { get; set; }
         public DateTime UpdatedDate { get; set; }
```
Comment (femhab) [inline LeaveApplication.Model/ViewModel/ResponseModel/UserInformationResponseModel.cs]: We might need to make updatedTime nullable

File: LeaveApplication.API/Controllers/LeaveApplicationController.cs (hunk 1 above)
Comment (AlakaTahir) [reply LeaveApplication.API/Controllers/LeaveApplicationController.cs:43]: I have renamed it to GetLeaveByUserId

File: LeaveApplication.API/Controllers/LeaveTypeController.cs (hunk 2 above)
Comment (AlakaTahir) [reply LeaveApplication.API/Controllers/LeaveTypeController.cs:52]: This is now handled

File: LeaveApplication.Model/ViewModel/ResponseModel/UserInformationResponseModel.cs (hunk 3 above)
Comment (AlakaTahir) [reply LeaveApplication.Model/ViewModel/ResponseModel/UserInformationResponseModel.cs]: This is now handled

**Review by femhab — APPROVED** [review]

**Pull request merged**